tt_eager/tt_dnn/op_library/sdpa/sdpa_op.cpp

```diff
@@ -233,6 +233,183 @@ tt::stl::reflection::Attributes ScaledDotProductAttention::attributes() const {
         {"valid_seq_len", this->valid_seq_len.value_or(0)}};
 }
 
+void ScaledDotProductAttentionDecode::validate(
+    const std::vector<Tensor>& input_tensors,
+    const std::vector<std::optional<const Tensor>>& optional_input_tensors) const {
+    TT_FATAL(input_tensors.size() == 3 and optional_input_tensors.size() == 1, "Must have 3 input tensors and mask");
+
+
+
+    for (auto& input_tensor : input_tensors) {
+        TT_FATAL(input_tensor.storage_type() == StorageType::DEVICE, "Operands to softmax need to be on device!");
+        TT_FATAL(input_tensor.buffer() != nullptr, "Operands to softmax need to be allocated in buffers on device!");
+        TT_FATAL((input_tensor.get_layout() == Layout::TILE), "Inputs to softmax must be tilized");
```

Comment:
Softmax?

Reply:
This is referring to the softmax within the kernel. Flash decode has a softmax component. Actually, this is copied from the flash-attention op. @cglagovichTT did you write it on purpose?

Reply:
I did not write that on purpose, I think it should be removed

Reply:
okay. I will change it for both sdpa and flash decode